arduino-ide-extension/src/node/core-service-impl.ts

```diff
@@ -379,21 +437,25 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
     port,
   }: {
     fqbn?: string | undefined;
-    port?: Port | undefined;
+    port?: PortIdentifier;
   }): Promise<void> {
-    this.boardDiscovery.setUploadInProgress(true);
-    return this.monitorManager.notifyUploadStarted(fqbn, port);
+    if (fqbn && port) {
```

Comment:
in which scenario would this be invoked without an `fqbn` & `port`?

Reply:
A long time ago, we decided (https://github.com/arduino/arduino-pro-ide/issues/364) not to make any premature args validation before running core CLI commands: verify, upload, burn the bootloader, etc. https://github.com/arduino/arduino-ide/commit/2831acc5b5b9f0c8744c946f1721925aaf5f8c9c

There are two main reasons:
 - there are cases when an upload could be valid when there is no port,
 - IDE2 wants to show the verbatim error message from the CLI: missing FQBN, the platform is not installed, etc.

> in which scenario would this be invoked without an `fqbn` & `port`?

When a board is not selected, a platform is not installed, or one has detached the board, it's still possible to run a verify/upload.